### PR TITLE
2050 relative volume paths windows

### PR DESCRIPTION
**Pull request by mnowster**
Make relative volume paths work on windows.

### CODE REVIEW

File: compose/const.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import sys
 
 DEFAULT_TIMEOUT = 10
+IS_WINDOWS_PLATFORM = (sys.platform == "win32")
```
Comment (dnephin) [inline compose/const.py]: this is now duplicated with line 12, we should remove one of them

Comment (mnowster) [inline compose/const.py]: Dammit. Good catch, thanks, I missed that in the rebase.

File: compose/service.py [hunk 2]
```diff
@@ -937,7 +938,19 @@ def build_volume_binding(volume_spec):
...
+    parts to be returned as a valid VolumeSpec tuple.
+    """
     parts = volume_config.split(':')
+
+    if IS_WINDOWS_PLATFORM:
+        # relative paths in windows expand to include the drive, eg C:\
+        # so we join the first 2 parts back together to count as one
+        windows_parts = [":".join(parts[0:2])]
+        windows_parts.extend(parts[2:])
+        parts = windows_parts
```
Comment (dnephin) [inline compose/service.py]: Could we do this as part of the config module instead?  If we have config always return a proper `VolumeSpec()` we don't have to worry about it twice

File: compose/config/config.py [hunk 3]
```diff
@@ -541,12 +542,19 @@ def path_mappings_from_dict(d):
...
+            host = ":".join(parts[0:2])
+            container = ":".join(parts[2:])
+            return (container, host)
     else:
-        return (string, None)
+        if ':' in path:
+            host, container = path.split(':', 1)
+            return (container, host)
+
+    return (path, None)
```
Comment (dnephin) [inline compose/config/config.py]: This feels like it's likely to hit a bunch of edge cases.  Would this work ?

```python

drive, volume_config = os.path.splitdrive(volume_config)
if ':' in volume_config:
    (host, container) = string.split(':', 1)
    return (container, drive + host)
else:
    return (drive + volume_config, None)
```

File: compose/service.py (hunk 2 above)
Comment (mnowster) [inline compose/service.py]: I'd rather do that as a separate PR to this. This is making it work first without too big a change, then we can refactor and improve. I have a feeling there'll be some refactoring required with the other `volumes_from` PR as well.

File: compose/config/config.py (hunk 3 above)
Comment (mnowster) [inline compose/config/config.py]: Let's find out!

**Comment by dnephin** [comment]
appveyor is now setup to run on PRs, so you should be able to just push this branch with a new commit and have it run

**Comment by mnowster** [comment]
Ok, the other test failures on appveyor are separate to what's going on in here. So this is good for final review/merge.

File: tests/unit/config/config_test.py [hunk 4]
```diff
@@ -1138,7 +1138,21 @@ def test_expand_path_with_tilde(self):
         self.assertEqual(result, user_path + 'otherdir/somefile')
 
 
-@pytest.mark.xfail(IS_WINDOWS_PLATFORM, reason='paths use slash')
```
Comment (dnephin) [inline tests/unit/config/config_test.py]: The test failures are because this decorator was removed from `BuildPathTest`.

I don't think we should merge with a red build. Please re-add this `xfail` so the build is green.

File: compose/config/config.py [hunk 5]
```diff
@@ -541,12 +541,13 @@ def path_mappings_from_dict(d):
...
-        (host, container) = string.split(':', 1)
-        return (container, host)
+def split_path_mapping(volume_path):
+    drive, volume_config = os.path.splitdrive(volume_path)
+    if ':' in volume_config:
+        (host, container) = volume_config.split(':', 1)
+        return (container, drive + host)
     else:
-        return (string, None)
+        return (volume_path, None)
```
Comment (dnephin) [inline compose/config/config.py]: We stripped the drive from `volume_path`, don't we need to re-add it?

I guess in this case it should always be the empty string, since the container path can't be a windows path?

This is really just a question, not a blocking issue.

File: compose/service.py (hunk 2 above)
Comment (dnephin) [inline compose/service.py]: That's fair. In that case can we use the same `splitdrive()` logic here as well?

File: tests/unit/config/config_test.py (hunk 4 above)
Comment (mnowster) [inline tests/unit/config/config_test.py]: Dammit, I fucked up the rebase and didn't spot this.

I know we don't merge on red build, I had assumed it was something else.

File: compose/config/config.py (hunk 5 above)
Comment (mnowster) [inline compose/config/config.py]: `volume_path` doesn't get changed, it's the result of `splitdrive` that we use. So rather than concating the two together again, we can return the `volume_path` as it was passed in.

Comment (dnephin) [inline compose/config/config.py]: Ahh, I misread sorry, you are right.  In my example I had re-used the variable, and I didn't read this properly.

**Comment by mnowster** [comment]
I'm falling into a pile of yaks with a chainsaw.
